### PR TITLE
Move easycrypt extraction to a separate executable

### PR DESCRIPTION
**Pull request by cassiersg**
Second step of the "extraction" plan.

Open question: which name for the new tool? (I picked `jasmin-ec`, but I'm bad at naming.)

cc @lyonel2017 @bgregoir

### CODE REVIEW

**Comment by bgregoir** [comment]
I think it is a good name.

**Comment by vbgl** [comment]
What about `jasmin2ec`?

File: compiler/src/main_compiler.ml [hunk 1]
```diff
@@ -171,25 +171,10 @@ let main () =
...
-        | fnames -> fnames in
-      begin try
-        BatPervasives.finally
-          (fun () -> close ())
-          (fun () -> ToEC.extract Arch.reg_size Arch.asmOp fmt !model prog fnames)
-          ()
-      with e ->
-        BatPervasives.ignore_exceptions
-          (fun () -> if !ecfile <> "" then Unix.unlink !ecfile) ();
-        raise e end;
```
Comment (vbgl) [inline compiler/src/main_compiler.ml:192]: Please leave this code here. The deprecation notice can go in the `CLI_errors` module.

It is wrong to use the `Unix` module in `ToEC`.

Comment (cassiersg) [reply compiler/src/main_compiler.ml]: done

**Comment by cassiersg** [comment]
@bgregoir `jasmin-ec` or `jasmin2ec` ?

File: compiler/src/toEC.ml [hunk 2]
```diff
@@ -1329,9 +1329,15 @@ let toec_prog pd asmOp model globs funcs arrsz warrsz randombytes =
...
-    Sint.iter (pp_array_decl ~prefix) !(env.arrsz);
-    Sint.iter (pp_warray_decl ~prefix) !(env.warrsz);
+    begin
+      match array_dir with
+    | Some prefix ->
+        begin
+          Sint.iter (pp_array_decl ~prefix) !(env.arrsz);
+          Sint.iter (pp_warray_decl ~prefix) !(env.warrsz)
+        end
+    | None -> ()
```
Comment (vbgl) [inline compiler/src/toEC.ml:1339]: This is a breaking change. Please document it (e.g., in the changelog).

**Comment by cassiersg** [comment]
Renamed to `jasmin2ec`.

**Review by vbgl — APPROVED** [review]
Is the reformatting of the CI file intended?

**Comment by cassiersg** [comment]
No, that was auto-formatting by my text editor that I forgot to disable.

**Review by vbgl — APPROVED** [review]
Thanks!

**Pull request merged**